src/main/java/teammates/storage/sqlapi/AccountRequestDb.java

```diff
@@ -0,0 +1,140 @@
+package teammates.storage.sqlapi;
+
+import java.time.Instant;
+import java.util.List;
+
+import org.hibernate.Session;
+
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.HibernateUtil;
+import teammates.storage.sqlentity.AccountRequest;
+
+import jakarta.persistence.TypedQuery;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
+
+/**
+ * Generates CRUD operations for AccountRequest.
+ *
+ * @see AccountRequest
+ */
+public final class AccountRequestDb extends EntitiesDb<AccountRequest> {
+    private static final AccountRequestDb instance = new AccountRequestDb();
+
+    private AccountRequestDb() {
+        // prevent instantiation
+    }
+
+    public static AccountRequestDb inst() {
+        return instance;
+    }
+
+    /**
+     * Creates an AccountRequest in the database.
+     */
+    public AccountRequest createAccountRequest(AccountRequest accountRequest)
+            throws InvalidParametersException, EntityAlreadyExistsException {
+        assert accountRequest != null;
+
+        accountRequest.sanitizeForSaving();
```

Comment:
We have moved the sanitisation methods to the setters instead.

Reply:
I have moved the sanitisation methods to the setters.